client/src/Auth/TabPanel.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from "react";
+import Typography from "@material-ui/core/Typography";
+import TextField from "@material-ui/core/TextField";
+import { Button } from "@material-ui/core";
+
+import { useStyles } from "./Auth.components";
+
+interface Props {
+  activeTab: number;
+  index: number;
+}
+
+const TabPanel: React.FC<Props> = ({ activeTab, index }) => {
+  const classes = useStyles();
+
+  const [credentials, setCredentials] = useState({
```

Comment:
you can unpack credentials here. It will save you creating intermediating variables in lines 25 to 28